stm/stm.ml

```diff
@@ -152,7 +197,14 @@ let mkTransCmd cast cids ceff cqueue =
 let summary_pstate = [ Evarutil.meta_counter_summary_name;
                        Evd.evar_counter_summary_name;
                        "program-tcc-table" ]
-type cached_state =
+type cached_state = {
+  mutable parsing : Parser.state;
+    (* Mutable because in some cases we we can't produce it without executing
+     * the command, but execution needs to add a node to the dag, and to
+     * build a node we give the parsing state :-/ *)
```

Comment:
I think this could be improved, see below. The point I don't like about the current solution is that it implies that at some point the contents of `parsing` will indeed be wrong for a while. So I'd better allow for the nodes to be built without a parsing state and update it at once, similarly to cached_state.